tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -628,6 +733,46 @@ policy_refresh_cagg_add_internal(Oid cagg_oid, Oid start_offset_type, NullableDa
 	JsonbValue *result = pushJsonbValue(&parse_state, WJB_END_OBJECT, NULL);
 	Jsonb *config = JsonbValueToJsonb(result);
 
+	CAggOverlapResult res = policy_refresh_cagg_check_for_overlaps(cagg, config, 0);
+	switch (res)
+	{
+		case CAGG_OVERLAP_NONE:
+			break;
+		case CAGG_OVERLAP_EXACT:
+			if (if_not_exists)
+			{
+				ereport(NOTICE,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("an identical continuous aggregate policy already exists for "
+								"\"%s\", "
+								"skipping",
+								get_rel_name(cagg->relid)),
+						 errdetail("A policy with the same start and end offset already exists for "
+								   "continuous aggregate \"%s\"",
+								   get_rel_name(cagg->relid))));
+				PG_RETURN_INT32(-1);
+			}
+			else
+			{
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("an identical continuous aggregate policy already exists for "
+								"\"%s\", ",
+								get_rel_name(cagg->relid)),
+						 errdetail("A policy with the same start and end offset already exists for "
+								   "continuous aggregate \"%s\"",
+								   get_rel_name(cagg->relid))));
+			}
```

Comment:
```suggestion

				ereport(ERROR,
						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
						 errmsg("an identical continuous aggregate policy already exists for "
								"\"%s\", ",
								get_rel_name(cagg->relid)),
						 errdetail("A policy with the same start and end offset already exists for "
								   "continuous aggregate \"%s\"",
								   get_rel_name(cagg->relid))));
```
Don't need the `else` branch.